### PR TITLE
Avoid pulling in magenta deps when building pip package

### LINKED ISSUE #317: Building the pip package shouldn't require installing Magenta dependencies
Currently, building the pip package requires installing Magenta dependencies like pretty_midi because setup.py does an `import magenta` to get the version number. We should move the version number into a separate file (version.py?) that is then referenced by both magenta/__init__.py and setup.py

### PR DESCRIPTION
**Pull request by cghawthorne**
Fixes #317

Also regenerate the main magenta dependencies.

### CODE REVIEW

**Comment by cghawthorne** [comment]
This is a little hacky, so let me know if you have thoughts about a better way to do it.

File: magenta/pipelines/BUILD [hunk 1]
```diff
@@ -135,8 +135,6 @@ py_library(
     srcs = ["pipelines_common.py"],
     deps = [
         ":pipeline",
-        "//magenta/music:chord_symbols_lib",
-        "//magenta/music:chords_lib",
         "//magenta/music:melodies_lib",
```
Comment (iansimon) [inline magenta/pipelines/BUILD]: You can also get rid of the melodies_lib dependency; not sure how I missed all of these.

File: magenta/tools/pip/setup.py [hunk 2]
```diff
@@ -16,7 +16,10 @@
 from setuptools import find_packages
 from setuptools import setup
 
-import magenta
+# Bit of a hack to parse the version string stored in version.py without
+# executing __init__.py, which will end up requiring a bunch of dependencies to
+# execute (e.g., tensorflow, pretty_midi, etc.).
+eval(compile(open('magenta/version.py').read(), 'magenta/version.py', 'exec'))
```
Comment (iansimon) [inline magenta/tools/pip/setup.py]: This is similar to #3 here: https://packaging.python.org/single_source_version/

File: magenta/pipelines/BUILD (hunk 1 above)
Comment (cghawthorne) [inline magenta/pipelines/BUILD]: Done

File: magenta/tools/pip/setup.py (hunk 2 above)
Comment (cghawthorne) [inline magenta/tools/pip/setup.py]: Oh neat, I didn't know about execfile. Switched to using that.

**Pull request merged**